src/main/java/htsjdk/samtools/seekablestream/SeekableStream.java

```diff
@@ -89,6 +89,28 @@ public void readFully(byte b[]) throws IOException {
         }
     }
 
+    /**
+     * Computes how many bytes are available in the stream.
```

Comment:
Javadoc for an overridden method isn't necessary as documentation is inherited. It can even make things worse, since any updates to the base class documentation wouldn't be reflected when viewing the javadoc for the overridden method.

Reply:
So do you suggest to add a `{@inheritDoc}` at the top? I just added the javadoc to show the concrete implementation...

Reply:
No javadoc is necessary, the base class doc will be used if no javadoc comment is provided. `@inheritdoc` is useful if you want to quote the entire base class doc and add more docs to it. In the case where you just want the base class doc, it's better to omit the javadoc completely.

Reply:
This is a different javadoc, that's why I am asking...

Reply:
Sure, but why is it different? Does it change the API as far as the caller is concerned?

I think the right way to think about javadoc is that it's documentation for the client of the API (the caller). Implementation details should by nature be not a client concern, and don't need to go in a javadoc. They would be more suited for an inline comment, e.g.
```
public int available() throws IOException {
    // If we are at the eof, 0 bytes are available.
    if (eof()) {
        return 0;
    }
    // Otherwise, available bytes are the difference between the length of the
    // stream and the current position.
    final long remaining = length() - position();
    // remaining might be negative if the length is not available, so return 0.
    return Math.max(0, (int) remaining);
}
```

Reply:
I understand your point here, but as an interface/abstract class it should point that implementations should be aware of this behavior... otherwise, it can have unexpected failures (I guess).

Anyway, I can just remove it if it's necessary for accepting this (there's not `@implNote` considered in HTSJDK yet, that would be ideal for this case)

Reply:
Looking at this class more closely I see that there are already many comments that replace the base class documentation, so I guess there's no harm in continuing the practice here.

Reply:
The "API" for `available()` is a complete disaster: "an estimate of the number of bytes that can be read without blocking", "returns 0 at end of stream, but also possibly at other times as well".

We should just override the javadoc to say "the return value of this method is unusable for any purpose, and we are only implementing it because certain Java classes like GZIPInputStream incorrectly rely on it to detect EOF"  :)